src/stores/WalletStore.ts

```diff
@@ -38,24 +36,39 @@ class WalletStore extends PersistableStore {
     return web3Modal.cachedProvider
   }
 
+  private async getUserNetworkName() {
+    return (await provider.getNetwork()).name
+  }
+
+  private async isNetworkRight() {
+    const network = env.VITE_ETH_NETWORK
+    const userNetwork = await this.getUserNetworkName()
+    return network === userNetwork
+  }
+
   async connect(clearCachedProvider = false) {
```

Comment:
Btw, looks like we use this as `true` usually, please check it and chagne if it is

Reply:
Will check, ty

Reply:
Actually, I don't think that it's a good idea to clear provider when it's not strictly defined, so I'll not change it to true as default